tests/test_BaseRegTest.py

```diff
@@ -90,11 +90,14 @@ def test_train_then_test_root(self):
         Also tests read/write in the process
         """
         self.ref_file = os.path.join(baseDir, "test_root.ref")
+        metadata = {"options": None}
         with BaseRegTest(self.ref_file, train=True) as handler:
             self.regression_test_root(handler)
+            handler.add_metadata(metadata)
```

Comment:
Can you add also another entry later i.e. after the first is added, then add lets say a version
```
handler.add_metadata(metadata)
# More added at later stage, like version
metadata["version"] = 1.2.3
handler.add_metadata(metadata)
```
I think the assert later is probably fine but check it

Reply:
I added this, but keep in mind this will only work if the second time `metadata` still contained the first entry. Otherwise I think this will overwrite the previous entry. I _can_ make it update the metadata field but maybe this is a more "expected" behaviour.